hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientMessage.java

```diff
@@ -412,21 +412,16 @@ public static ClientMessage create() {
     }
 
     public static ClientMessage createForEncode(int initialCapacity) {
-        initialCapacity = findSuitableMessageSize(initialCapacity);
+        if (initialCapacity < 0) {
+            throw new MaxMessageSizeExceeded();
+        }
         if (USE_UNSAFE) {
             return createForEncode(new UnsafeBuffer(new byte[initialCapacity]), 0);
         } else {
             return createForEncode(new SafeBuffer(new byte[initialCapacity]), 0);
         }
     }
 
-    public static int findSuitableMessageSize(int desiredMessageSize) {
```

Comment:
This method is public static, was it designed to be used by user? I hope not, since it's in the impl package, but I would like some confirmation that it's ok to remove it, e.g. @sancar ?

Reply:
I see no problem here. It is ok to remove.